### PR TITLE
pool: Drop opening no-op sessions

### PR DESCRIPTION
**Pull request by cthulhu-rider**

### CODE REVIEW

File: pool/object_test.go [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+}
+
+type objectSearchOnlyClientWrapper struct {
+	mockedClientWrapper
+	c objectSearchOnlyClient
+}
+
+func (x objectSearchOnlyClientWrapper) getClient() (sdkClientInterface, error) { return x.c, nil }
+
+func TestPool_ObjectSearchInit(t *testing.T) {
```
Comment (roman-khimov) [inline pool/object_test.go:507]: Why duplicating prologue in these? Likely we can have less code and still separate subtests for different operations.

**Review by smallhive — COMMENTED** [review]
It is a feature of the pool that allows autosession. We removing it here, do we need to handle sessions ourselves again?

**Comment by roman-khimov** [comment]
Sessions are still there where they're needed. It's just that most requests can be handled fine without sessions.

File: pool/object_test.go (hunk 1 above)
Comment (cthulhu-rider) [reply pool/object_test.go:507]: prefer independent tests for methods

**Review by smallhive — APPROVED** [review]

**Pull request merged**